src/brpc/policy/discovery_naming_service.cpp

```diff
@@ -271,9 +272,15 @@ int DiscoveryClient::DoRegister() {
     cntl.http_request().set_content_type("application/x-www-form-urlencoded");
     butil::IOBufBuilder os;
     os << "appid=" << _params.appid
-        << "&hostname=" << _params.hostname
-        << "&addrs=" << _params.addrs
-        << "&env=" << _params.env
+        << "&hostname=" << _params.hostname;
+
+    std::vector<butil::StringPiece> addrs;
+    butil::SplitString(_params.addrs, ',', &addrs);
+    for (size_t i = 0; i < addrs.size(); ++i) {
+        os << "&addrs=" << addrs[i];
+    }
```

Comment:
According to the comment of butil::SplitString: If several instances of |c| are contiguous, or if |str| begins with or ends with |c|, then an empty string is inserted, we should check if addrs[i] is empty string.

Reply:
Thanks for the review, and done.